packages/contracts-bedrock/test/fixtures/L2GenesisFixtures.sol

```diff
@@ -0,0 +1,290 @@
+    }
+
+    /// @notice Returns the expected hardcoded deployed bytecode for a given address. Depending on the
+    ///         constructor args/initialization args, the hardcoded bytecode may need to be updated to
+    ///         what's expected for your specific usecase.
+    function getExpectedDeployedBytecode(address _addr) public view returns(bytes memory) {
+        return expectedDeployedBytecode[_addr];
+    }
+
+    /// @notice Sets the above mappings (`numExpectedSlotKeys`, `expectedSlotKeys`,
+    ///         and `slotValueByKey`) for each predeploy proxy.
+    function _setProxyStorage() internal {
+        uint160 prefix = uint160(0x420) << 148;
+
+        for (uint256 i; i < L2GenesisHelpers.PROXY_COUNT; i++) {
+            address addr = address(prefix | uint160(i));
+
+            if (L2GenesisHelpers.notProxied(addr)) continue;
+
+            numExpectedSlotKeys[addr] = ++numExpectedSlotKeys[addr];
+            expectedSlotKeys[addr][EIP1967Helper.PROXY_OWNER_ADDRESS] = true;
+            slotValueByKey[addr][EIP1967Helper.PROXY_OWNER_ADDRESS] = bytes32(uint256(uint160(Predeploys.PROXY_ADMIN)));
+
+            /// @dev It's important this code execute after the above `L2GenesisHelpers._notProxied` check.
+            /// This is because WETH9 and GOVERNANCE_TOKEN are included in `L2GenesisHelpers._isDefinedPredeploy`,
+            /// but are NOT proxied and therefore shouldn't have the following storage slot set.
+            if (L2GenesisHelpers.isDefinedPredeploy(addr)) {
+                address implementation = L2GenesisHelpers.predeployToCodeNamespace(addr);
+                numExpectedSlotKeys[addr] = ++numExpectedSlotKeys[addr];
+                expectedSlotKeys[addr][EIP1967Helper.PROXY_IMPLEMENTATION_ADDRESS] = true;
+                slotValueByKey[addr][EIP1967Helper.PROXY_IMPLEMENTATION_ADDRESS] = bytes32(uint256(uint160(implementation)));
+            }
+        }
+
+        _setWETH9ProxyStorage();
+        _setL2CrossDomainMessengerProxyStorage();
+        _setL2StandardBridgeProxyStorage();
+        _setOptimismMintableERC20FactoryProxyStorage();
+        _setGovernanceTokenProxyStorage();
+    }
+
+    /// @notice Sets the `expectedDeployedBytecode` mapping for each predeploy implementation.
+    function _setImplementationStorage() internal {
+        _setLegacyMessagePasserImplStorage();
+        _setDeployerWhitelistImplStorage();
+        _setL2CrossDomainMessengerImplStorage();
+        _setGasPriceOracleImplStorage();
+        _setL2StandardBridgeImplStorage();
+        _setSequencerFeeWalletImplStorage();
+        _setOptimismMintableERC20FactoryImplStorage();
+        _setL1BlockNumberImplStorage();
+        _setL1BlockAttributesImplStorage();
+    }
+
+    //////////////////////////////////////////////////////
+    /// Predeploy Proxy Storage Setters
+    //////////////////////////////////////////////////////
+    function _setWETH9ProxyStorage() internal {
```

Comment:
This is not scalable or legible, we should use the `vm.stopAndReturnStateDiff()` related cheatcodes around the internal functions that setup individual contracts